File: Searching/Linear Search/linear_search.py
# Linear Search Algorithm in Python
# =================================================
def linear_search(arr, element):
    pos = 0
    for i in range(len(arr)):
        if arr[i] == element:
            pos = i + 1
            return True, pos
    return False, pos

def main():
    arr = [7,8,10,15,29]
    element = 10

    found, pos = linear_search(arr, element)
    if found:
        print("Element {} found at position {}".format(element, pos))
    else:
        print("No element found")

main()
# ================================================
# By Abel Roy #

File: Searching/Linear Search/linear_search.c
/* Linear Search Algorithm in C Lang */
// =================================================
#include<stdio.h>

void linear_search(int arr[],int element){
    int length = 5;
    int found = 0, pos = 0;

    for (int i=0; i<length; i++){
        if (arr[i] == element){
            found = 1;
            pos = i;
            break;
        }
    }

    if (found == 1)
        printf("Found element %d at position %d\n",element, pos+1);
    else
        printf("Element not found");
}

void main(){
    int arr[5] = {1,2,3,4,5};
    int element = 4;

    linear_search(arr, element);
}
// ================================================
/* By Abel Roy */
